app/Filament/Resources/ServerResource.php

```diff
@@ -103,6 +158,40 @@ public static function infolist(Infolist $infolist): Infolist
             ->schema(self::getServerInfoList());
     }
 
+    public static function testConnection(array $state): void
+    {
+        $credentials = ServerCredential::query()->find((int)$state["server_credentials_id"]);
+        $connection = new Connection(
+            $credentials?->ssh_private_key,
```

Comment:
Since a server can have multiple credentials this method should also take a credential_id as an input. Maybe if there is only a single credential we can use that, or perhaps we can just pick the first credential to test the server.